9 gateway microservices

KanID92 opened the pull request:
Sprint 22. Add:
- module "order"
- module "delivery"
- module "payment".

KanID92 committed:
Sprint 22 (v.0.8).

KanID92 committed:
Sprint 22 (v.1.0).
Add:
- module "delivery";
- module "order";
- module "payment".

kesch9 commented on commerce/order/src/main/java/ru/yandex/practicum/commerce/order/mapper/impl/GeneralAddressMapper.java:
```diff
@@ -0,0 +1,19 @@
+public class GeneralAddressMapper implements AddressMapper {
+    @Override
+    public Address toAddress(AddressDto addressDto) {
+        return null;
```
Обязательно
Если в мапперах ничего не делается для address и order, то они и не нужны

kesch9 commented on commerce/delivery/src/main/java/ru/yandex/practicum/commerce/delivery/service/impl/GeneralDeliveryService.java:
```diff
@@ -0,0 +1,101 @@
+    private final OrderClient orderClient;
+    private final WarehouseClient warehouseClient;
+
+    private final float baseCost = 5;
```
Хорошо вынести в настройки или сделать константой

KanID92 replied:
Вынес в application.yaml

kesch9 commented on commerce/delivery/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,30 @@
+
+    delivery_id             uuid DEFAULT gen_random_uuid(),
+    from_address_id         uuid NOT NULL,
+    to_address_id           uuid NOT NULL,
```
Обязательно
from_address_id и to_address_id это FK на addresses нужно добавить

KanID92 replied:
Откорректировано.

kesch9 commented on commerce/order/src/main/java/ru/yandex/practicum/commerce/order/OrderController.java:
```diff
@@ -0,0 +1,132 @@
+
+
+
+
```
Много лишних строк))

KanID92 replied:
Убрал)

kesch9 commented on commerce/order/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,51 @@
+    delivery_price double precision,
+    product_price double precision,
+    state varchar(50),
+    address_id uuid,
```
Также не хватает определения FK

KanID92 replied:
Откорректировано.

kesch9 reviewed (commented):
Парочка комментариев

KanID92 committed:
Sprint 22.
fix:
- mappers AddressMapper, OrderMapper (add impl and remove unnecessary);
- GeneralDeliveryService: config value moved to application.yaml;
- schema.sql: delivery and order - add foreign keys.

kesch9 reviewed (approved):
Approve
